testutils/src/main/java/com/amplifyframework/testutils/sync/SynchronousMobileClient.java

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.testutils.sync;
+
+import android.content.Context;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.test.core.app.ApplicationProvider;
+
+import com.amplifyframework.core.Consumer;
+import com.amplifyframework.testutils.Await;
+
+import com.amazonaws.mobile.client.AWSMobileClient;
+import com.amazonaws.mobile.client.Callback;
+import com.amazonaws.mobile.client.UserStateDetails;
+import com.amazonaws.mobile.client.results.SignInResult;
+import com.amazonaws.mobile.config.AWSConfiguration;
+
+import java.util.Objects;
+
+/**
+ * A wrapper around the {@link AWSMobileClient} which turns its asynchronous methods
+ * into synchronous ones.
+ */
+@SuppressWarnings("unused")
+public final class SynchronousMobileClient {
```

Comment:
There is an existing class called `TestMobileClient` (I think) and this is supposed to be a rename and iteration on top of that existing file

Reply:
https://github.com/aws-amplify/amplify-android/blob/master/aws-api/src/androidTest/java/com/amplifyframework/api/aws/TestAWSMobileClient.java

Is this what you are referring to? I believe `SynchronousMobileClient` class is more extensive and it should replace the old `TestAWSMobileClient` in other categories

Reply:
Oops I misunderstood your intent. Removing that file and replacing its usage with the new `SynchronousMobileClient` now